collector/lib/KernelDriver.h

```diff
@@ -141,8 +142,37 @@ class KernelDriverCOREEBPF : public IKernelDriver {
   KernelDriverCOREEBPF() = default;
 
   bool Setup(const CollectorConfig& config, sinsp& inspector) override {
-    // TODO: implement CO.RE ebpf setup if needed
-    return false;
+    /* Capture only necessary tracepoints and syscalls. */
+    auto tp_set = libsinsp::events::enforce_simple_tp_set();
+    std::unordered_set<ppm_sc_code> ppm_sc;
+
+    /*
+     * Convert text reprecentation of event type into an actual syscall code
+     * using g_syscall_table.
+     */
+    const EventNames& event_names = EventNames::GetInstance();
+    for (const auto& syscall_str : config.Syscalls()) {
+      for (ppm_event_type event_id : event_names.GetEventIDs(syscall_str)) {
+        uint16_t syscall_id = event_names.GetEventSyscallID(event_id);
+        if (!syscall_id) {
+          continue;
+        }
+
+        syscall_evt_pair syscall = g_syscall_table[syscall_id];
+        ppm_sc.insert((ppm_sc_code)syscall.ppm_sc);
+      }
+    }
```

Comment:
For custom ebpf probes it will most likely not change anything, because they already filter out syscalls based on the attachment points. For kernel modules it would make sense, although I'm pretty sure there is already some filtering happening inside.